<launch>
    <param name="seq" type="int" value="8" />
    <param name="save_directory" type="string" value="/home/y/evo/ev_ki/fusion/08/"/>  
    <node pkg="gtPose" type="alaserPGO" name="alaserPGO" output="screen" /> 


    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find orb_slam2_ros)/ros/config/rviz_config.rviz" />
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_orb_slam2" output="screen" ns="orb_slam2_ros" >
        <param name="/target_frame_name" value="/map" />
        <param name="/source_frame_name" value="/base_link" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node>

   <node name="orb_slam2_rgbd" pkg="orb_slam2_ros"
      type="orb_slam2_ros_rgbd" output="screen">

       <remap from="/camera/rgb/image_raw" to="/image_left"  />
       <remap from="/camera/depth_registered/image_raw" to="/image_depth_dense" />

	<param name="save_trajectoryTum" type="bool" value="true"/>
       <param name="publish_pointcloud" type="bool" value="true" />
       <param name="publish_pose" type="bool" value="true" />
       <param name="localize_only" type="bool" value="false" />
       <param name="reset_map" type="bool" value="false" />

       <!-- static parameters -->
       <param name="load_map" type="bool" value="false" />
       <param name="map_file" type="string" value="map.bin" />
       <param name="voc_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/Vocabulary/ORBvoc.txt" />

       <param name="pointcloud_frame_id" type="string" value="map" />
       <param name="camera_frame_id" type="string" value="camera_link" />
       <param name="min_num_kf_in_map" type="int" value="5" />

       <!-- ORB parameters -->
       <param name="/ORBextractor/nFeatures" type="int" value="1200" />
       <param name="/ORBextractor/scaleFactor" type="double" value="1.2" />
       <param name="/ORBextractor/nLevels" type="int" value="8" />
       <param name="/ORBextractor/iniThFAST" type="int" value="25" />
       <param name="/ORBextractor/minThFAST" type="int" value="10" />

       <!-- Camera parameters -->
       <!-- Camera frames per second -->
       <param name="camera_fps" type="int" value="10" />
       <!-- Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale) -->
       <param name="camera_rgb_encoding" type="bool" value="true" />
       <!-- Close/Far threshold. Baseline times. -->
       <param name="ThDepth" type="double" value="70.0" />
       <!-- Deptmap values factor (what pixel value in the depth image corresponds to 1m) -->
       <param name="depth_map_factor" type="double" value="255.0" />

        <!-- Camera calibration parameters -->
        <!--If the node should wait for a camera_info topic to take the camera calibration data-->
       <param name="load_calibration_from_cam" type="bool" value="false" />
       <!-- Camera calibration and distortion parameters (OpenCV) -->
      <param name="camera_fx" type="double" value="707.0912" />
      <param name="camera_fy" type="double" value="707.0912" />
      <param name="camera_cx" type="double" value="601.8873" />
      <param name="camera_cy" type="double" value="183.1104" />
       <!-- Camera calibration and distortion parameters (OpenCV) -->
      <param name="camera_k1" type="double" value="0.0" />
      <param name="camera_k2" type="double" value="0.0" />
      <param name="camera_p1" type="double" value="0.0" />
      <param name="camera_p2" type="double" value="0.0" />
      <param name="camera_k3" type="double" value="0.0" />
      <!-- IR projector baseline times fx (aprox.) -->
      <param name="camera_baseline" type="double" value="379.8145" />
  </node>

    <node pkg="tf" type="static_transform_publisher" name="image_base_broadcaster" args="0 0 0 0 0 0  /base_link /camera_link 100" />





    <param name="scan_line" type="int" value="64" />

    <!-- if 1, do mapping 10 Hz, if 2, do mapping 5 Hz. Suggest to use 1, it will adjust frequence automaticlly -->
    <param name="mapping_skip_frame" type="int" value="1" />

    <!-- remove too closed points -->
    <param name="minimum_range" type="double" value="5"/>

    <param name="mapping_line_resolution" type="double" value="0.4"/> <!-- A-LOAM -->
    <param name="mapping_plane_resolution" type="double" value="0.8"/> <!-- A-LOAM -->

    <!-- SC-A-LOAM -->
    <param name="keyframe_meter_gap" type="double" value="2.0"/> 

    <!-- Scan Context -->
    <param name="sc_dist_thres" type="double" value="0.4"/> <!-- SC-A-LOAM, if want no outliers, use 0.1-0.15 -->
    <!-- <param name="sc_max_radius" type="double" value="20.0"/> 20 or 40 for indoor -->
    <param name="sc_max_radius" type="double" value="80.0"/> <!-- for outdoor -->





  
    <param name="lidar_type" type="string" value="HDL64"/> 
    <remap from="/velodyne_points" to="/velodyne_points"/>

    <node pkg="aloam_velodyne" type="ascanRegistration" name="ascanRegistration" output="screen" />
    <node pkg="aloam_velodyne" type="alaserOdometry" name="alaserOdometry" output="screen" />
    <node pkg="aloam_velodyne" type="alaserMapping" name="alaserMapping" output="screen" />
   

</launch>
